tests/specs/lemmas.k

```diff
@@ -225,6 +226,14 @@ module LEMMAS-JAVA [symbolic, kast]
 
     rule chop(I) => I requires 0 <=Int I andBool I <Int pow256  [simplification]
 
+    //Simplification of bool2Word() ==Int CONCRETE, #buf() ==K CONCRETE
+    rule I                   ==K bool2Word( B:Bool ) => bool2Word(B) ==Int I    requires #isConcrete(I)  [simplification]
+    rule bool2Word( B:Bool ) ==K I                   => B ==K word2Bool(I)      requires #isConcrete(I)  [simplification]
+
+    rule BA:ByteArray     ==K #buf( 32, DATA ) => #buf( 32, DATA ) ==K BA       requires #isConcrete(BA) [simplification]
+    rule #buf( 32, DATA ) ==K BA:ByteArray     => DATA ==Int #asInteger(BA)
+      requires #sizeByteArray(BA) <=Int 32                                      andBool  #isConcrete(BA) [simplification]
+
```

Comment:
I think this is the nontrivial change in this PR.

@denis-bogdanas Could you please explain why this is needed? I mean, why the previous lemmas that you are going to delete above are not enough?